runtime/compiler/x/codegen/CheckFailureSnippet.cpp

```diff
@@ -149,10 +149,10 @@ uint8_t *TR::X86CheckFailureSnippet::emitCheckFailureSnippetBody(uint8_t *buffer
       *buffer++ = 0xd8;
       }
 
-   *buffer++ = 0xe8; // CallImm4
+   *buffer = 0xe8; // CallImm4
    intptr_t destinationAddress = (intptr_t)getDestination()->getMethodAddress();
 
-   if (NEEDS_TRAMPOLINE(destinationAddress, buffer+4, cg()))
+   if (cg()->directCallRequiresTrampoline(destinationAddress, reinterpret_cast<intptr_t>(buffer++)))
```

Comment:
Is this correct? I thought you pass in the start of the branch instruction to `directCallRequiresTrampoline`? As it stands, aren't you passing in the address of the immediate?

Reply:
No, that's a post-increment operator on `buffer` (it is updated after the expression is evaluated).